File: mapir-android-service-sdk/src/main/java/ir/map/servicesdk/response/SearchResponse.java
package ir.map.servicesdk.response;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import ir.map.servicesdk.model.base.MapirResponse;
import ir.map.servicesdk.model.inner.SearchItem;

import static ir.map.servicesdk.MapService.createGeom;

public class SearchResponse extends MapirResponse {

    private Integer count;
    private List<SearchItem> searchItems;

    private SearchResponse(int count, List<SearchItem> searchItems) {
        this.count = count;
        this.searchItems = searchItems;
    }

    public Integer getCount() {
        return count;
    }

    public List<SearchItem> getSearchItems() {
        return searchItems;
    }

    public static MapirResponse createSearchResponse(String data) {
        try {
            JSONObject tempData = new JSONObject(data);
            JSONArray tempSearchItems = new JSONArray(tempData.get("value").toString());

            List<SearchItem> searchItems = new ArrayList<>();
            for (int i = 0; i < tempSearchItems.length(); i++) {
                JSONObject tempSearchItem = new JSONObject(tempSearchItems.get(i).toString());

                SearchItem item = new SearchItem(
                        tempSearchItem.getString("province"),
                        tempSearchItem.getString("county"),
                        tempSearchItem.getString("district"),
                        tempSearchItem.getString("city"),
                        tempSearchItem.getString("region"),
                        tempSearchItem.getString("neighborhood"),
                        tempSearchItem.getString("title"),
                        tempSearchItem.getString("address"),
                        tempSearchItem.getString("type"),
                        tempSearchItem.getString("fclass"),
                        createGeom(tempSearchItem.get("geom").toString())
                );

                searchItems.add(item);
            }

            return new SearchResponse(
                    tempData.getInt("odata.count"),
                    searchItems
            );
        } catch (Exception e) {
            return null;
        }
    }
}

File: mapir-android-service-sdk-ktx/src/main/java/ir/map/servicesdk/response/SearchResponse.kt
package ir.map.servicesdk.response

import ir.map.servicesdk.MapService.Companion.createGeom
import ir.map.servicesdk.model.base.MapirResponse
import ir.map.servicesdk.model.inner.SearchItem
import org.json.JSONArray
import org.json.JSONObject
import java.util.*

class SearchResponse(val count: Int, val searchItems: List<SearchItem>) : MapirResponse() {
    companion object {
        fun createSearchResponse(data: String) = try {
            val tempData = JSONObject(data)
            val tempSearchItems = JSONArray(tempData["value"].toString())
            val searchItems: MutableList<SearchItem> = ArrayList()

            for (i in 0 until tempSearchItems.length()) {
                val tempSearchItem = JSONObject(tempSearchItems[i].toString())
                val item = SearchItem(
                        tempSearchItem.getString("province"),
                        tempSearchItem.getString("county"),
                        tempSearchItem.getString("district"),
                        tempSearchItem.getString("city"),
                        tempSearchItem.getString("region"),
                        tempSearchItem.getString("neighborhood"),
                        tempSearchItem.getString("title"),
                        tempSearchItem.getString("address"),
                        tempSearchItem.getString("type"),
                        tempSearchItem.getString("fclass"),
                        createGeom(tempSearchItem["geom"].toString())!!
                )

                searchItems.add(item)
            }

            SearchResponse(
                    tempData.getInt("odata.count"),
                    searchItems
            )
        } catch (e: Exception) {
            null
        }
    }
}